Virology update - right first time edition 2 electric boogaloo

LanCartwright opened the pull request:
:cl:
Adds var "needs_all_cures" to viruses. Default TRUE.

Adds to advanced symptoms:
Asthmosthia. A atmospheric gas causing virus. Very noticable.
Scalerot. A symptom that only affects lizards.
Sensory restoration and destruction. Effects are it's titular namesake.
Adds viral adaptation. A virus that massively buffs viruses stealth and resistance in return for slower stage speeds.

Adds to viruses:
Bubonic scabies. A virus that affects humans only.
Dox Pox. A self curing virus that prevents equipping outside clothing.
Kilkof Hemorrhagic fever. A deadly, extremely slow acting virus that kills upon it's 21st stage.
Andromeda blobitis. A deadly, slow acting virus that kills the victim and turns them into a blob zombie.
Skin mutation. A non-lizard only virus that turns non-lizards into lizards.
Scarabas affiction. A highly dangerous, infectious virus that causes the host to spawn scarabitic cockroaches that're friendly to them.

Adds to suits/armour:
Pox armour - used for Dox pox virus.

Adds to species types:
Scarabite. A disguised, armoured human-bug hybrid. Immune to damaging effects of Scarabas affliction and far more powerful than humans.

Adds to giant spider:
Scarabite cockroach. A smarter, stronger, deadlier cockroach spawned by scarabaras afflicted persons. Will actively infect others.

Adds to other reagents:
Scarabite egg venom. Infects victim with scarabite affliction.
Lye now damages people.

Changes to symptoms:
Spontaneous combustion
Stealth: -4 to -3
Resistance: -4 to -3

Necrotizing Fasciitis
Resistance: -4 to -3
Transmittance: -4 to -2

Sneezing
Stealth:  -2 to -1
Resistance:  3 to 2

Hyphema
Resistance: -4 to -2

Eternal Youth
Resistance: 4 to 5
Stage speed: 4 to 5

/:cl:

GunHog commented on code/datums/diseases/advance/symptoms/sneeze.dm:
```diff
@@ -19,8 +19,8 @@ Bonus
 	name = "Sneezing"
-	stealth = -2
-	resistance = 3
+	stealth = -1
```
Why are you decreasing stealth rating? Sneezing symptom is a useful tool for revealing stealth virus mixtures in the PANDEMIC. If you are doing this to make it more useful in combat viruses, please provide a suitable roundstart replacement. Perhaps give -2 stealth to cough instead?

PKPenguin321 replied:
Yeah this is shit

Shadowlight213 replied:
@Gunhog unless I'm mistaken, I'm pretty sure this is actually decreasing the stealth penalty that the symptom gives, not increasing the penalty. It's a buff. Positive values improve stealth/transmission/resistance/etc.. while negative values decrease it.

Giacom replied:
He meant increasing but his point is still valid.

RemieRichards commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,64 @@
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+		var/mob/living/M = A.affected_mob
+		if(human.dna.species.id == "lizard")//this is scale rot. literally no other race has scales, please contact me if more scaled species are added in
+		switch(A.stage)
```
this switch isn't in the scope of the If(), so the if does nothing.

RemieRichards commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,64 @@
+			if(5)
+				if(M.dna.species.id != "lizard")
+					stage = 6
+				if(ishuman(M)
```
missing )

RemieRichards commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,64 @@
+	..()
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+		var/mob/living/M = A.affected_mob
+		if(human.dna.species.id == "lizard")//this is scale rot. literally no other race has scales, please contact me if more scaled species are added in
```
"human"? there's no human var in scope here.

neersighted replied:
Should be `A`.

RemieRichards replied:
no... no it shouldn't.

neersighted replied:
Oh you.

RemieRichards commented on code/datums/diseases/lizarditis.dm:
```diff
@@ -0,0 +1,53 @@
+	..()
+	var/mob/living/M = A.affected_mob
+	if(M.dna.species.id != "lizard")
+	switch(stage)
```
this switch isn't in the scope of the if, so the if does nothing.

neersighted commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,64 @@
+				M << "<span class='notice'>[pick("Your scales rot away and reveal flesh")]</span>"
+				M.adjustcloneLoss(45)
+			if(5)
+				if(M.dna.species.id != "lizard")
```
All these checks are bad, they should be moved up scopewise.

neersighted commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,64 @@
+						human.update_body()
+						human.update_hair()
+						M.adjustcloneLoss(200)
+						M.adjustbruteLoss(200)
```
Its `adjustBruteLoss` and `adjustCloneLoss`

neersighted commented on code/datums/diseases/advance/symptoms/sensory_restoration.dm:
```diff
@@ -0,0 +1,51 @@
+			if(4, 5)
+				if (M.reagents.get_reagent_amount("mannitol") < 15)
+					M.reagents.add_reagent("mannitol", 15)
+			if(prob(SYMPTOM_ACTIVATION_PROB * 5))
```
Ifs in a case must be constant

neersighted commented on code/datums/diseases/kilkofhemorrhage.dm:
```diff
@@ -0,0 +1,97 @@
+						affected_mob << "<span class='userdanger'>You go blind!</span>"
+						affected_mob.disabilities |= BLIND
+						affected_mob.eye_blind = 1
+			if(prob(10))
```
It would probably be better to rand() and then switch based on it. Saves all these prob() calls.

neersighted commented on code/datums/diseases/lizarditis.dm:
```diff
@@ -0,0 +1,53 @@
+				src.cure()
+			M << "<span class='notice'>[pick("Your skin feels awfully itchy", "Something tries to stick out of your jumpsuit")]</span>"
+		if(3)
+			if(M.dna.species.id == "lizard")
```
Move all these checks out of scope.

Lumipharon commented:
could we get a brief description of what all the actual changes are in the OP? Just like changing the stealth rating are fairly obvious, other things less so.

GunHog commented:
Did you **compile and test** this before you put it up for PR? I like most of the PR (Except the lizard virus and changes to Sneeze) and would like to see it merged, but you have made some errors that should not make it past your own testing.

RemieRichards commented on code/datums/diseases/lizarditis.dm:
```diff
@@ -0,0 +1,53 @@
+
+/datum/disease/lizarditis/stage_act()
+	..()
+	var/mob/living/M = A.affected_mob
```
No "A" in scope.

neersighted commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,63 @@
+/datum/symptom/scalerot/Activate(var/datum/disease/advance/A)
+	..()
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+	var/mob/living/mob/living/carbon/human/M = affected_mob
```
Needs to be indented or the if will have no effect.

RemieRichards replied:
>var/mob/living/mob/living/carbon/human/H

neersighted replied:
`var/mob/living/mob/living` kek

neersighted commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,63 @@
+	var/mob/living/mob/living/carbon/human/M = affected_mob
+		switch(A.stage)
+			if(1)
+				if(M.dna.species.id != "lizard")
```
Please move all these checks out

neersighted commented on code/datums/diseases/advance/symptoms/sensory_restoration.dm:
```diff
@@ -0,0 +1,53 @@
+*/
+
+/datum/symptom/sensory_restoration
+
```
No newlines please

neersighted commented on code/datums/diseases/advance/symptoms/viraladaptation.dm:
```diff
@@ -0,0 +1,36 @@
+*/
+
+/datum/symptom/viraladaptation
+
```
No newline here

neersighted commented on code/datums/diseases/kilkofhemorrhage.dm:
```diff
@@ -0,0 +1,99 @@
+					if(I && I.w_class == 1)
+						affected_mob.drop_item()
+		if(16, 17, 18, 19, 20)
+			switch(rand(1,6))
```
Spaces in proc calls

neersighted commented on code/datums/diseases/kilkofhemorrhage.dm:
```diff
@@ -0,0 +1,99 @@
+			affected_mob.visible_message("<span class='warning'>[affected_mob] seizes up and begins violently shaking, as if he's going into shock!</span>")
+			affected_mob.visible_message("<span class='warning'>[affected_mob] begins a chilling deathrattle, as if he's going into cardiac arrest!</span>")
+			affected_mob.visible_message("<span class='warning'>[affected_mob] begins to foam at the mouth, as if his stomach is releasing acid!</span>")
+			if (affected_mob.reagents.get_reagent_amount("initropidril, mutagen, lexorin, venom, pancuronium, curare, coniine, sacid") < 50)
```
No spaces in if(

neersighted commented on code/datums/diseases/kilkofhemorrhage.dm:
```diff
@@ -0,0 +1,99 @@
+			affected_mob.visible_message("<span class='warning'>[affected_mob] begins a chilling deathrattle, as if he's going into cardiac arrest!</span>")
+			affected_mob.visible_message("<span class='warning'>[affected_mob] begins to foam at the mouth, as if his stomach is releasing acid!</span>")
+			if (affected_mob.reagents.get_reagent_amount("initropidril, mutagen, lexorin, venom, pancuronium, curare, coniine, sacid") < 50)
+				affected_mob.reagents.add_reagent_list(list("initropidril" = 50, "mutagen" = 50, "lexorin" = 50, "venom" = 50, "pancuronium" = 50, "curare" = 50, "coniine" = 50, "sacid" = 50))
```
Indentation error?

neersighted commented on code/datums/diseases/advance/symptoms/vision.dm:
```diff
@@ -19,7 +19,7 @@ Bonus
 
```
newlines

neersighted commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,63 @@
+				M << "<span class='notice'>[pick("Your scales rot away and reveal flesh")]</span>"
+				M.adjustcloneLoss(45)
+			if(5)
+				if(M.dna.species.id != "lizard")
```
Seriously get all these checks out of here

neersighted commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,63 @@
+						human.update_body()
+						human.update_hair()
+						M << "<span class='notice'>[pick("You tear at your scales and rip off your scaley skin!")]</span>"
+						M.adjustCloneLoss(200)
```
You know I'm pretty sure we have a MAX_HEALTH define. Also, doing 400 damage is a bit much, 200 will kill outright.

neersighted commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,63 @@
+*/
+
+/datum/symptom/scalerot
+
```
reeee newlines

neersighted commented on code/datums/diseases/advance/symptoms/asthmosthia.dm:
```diff
@@ -0,0 +1,40 @@
+*/
+
+/datum/symptom/asthmothia
+
```
newline

KorPhaeron commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,63 @@
+				M.adjustcloneLoss(15)
+			if(3)
+				if(M.dna.species.id != "lizard")
+					stage = 6
```
>copypasting the same check 5 times

please

Zilenan91 commented:
I'd still like scalerot removed or made to not kill people at the very least.

LanCartwright commented:
I'll consider it. Scalerot is a deadly, deadly virus to lizard and snake pets in real life hence why I based it so deadly.

KazeEspada commented:
Bubonic plague can literally floor an entire continent and then some more. However, Lizards and snakes can not carry the disease. Maybe we should add that?

optimumtact commented:
hahah maybe we should make a virust that immediately stuns and kills you and is present at roundstart in everyone

that sounds like a lot of fun

LanCartwright commented:
You know what. Just for you Kaze, I will.

I'll add that.

Zilenan91 commented:
How about we don't make diseases that target specific species?

Lumipharon commented:
I don't see the problem with that, tbh.
They're ony usable by antags without bannu/unfortunate "accidents", so it's not really a problem.

Having lots of extra symptoms added with the way viro works will make getting specific symptoms more tedious, however.

Viro would be infinitely better if the pandemic/new machine could simply remove specific symptoms, rather then the frankly time wasting and pointless rng removal we have at the moment.

Dorsisdwarf commented:
release skin mutation + scale rot, kill everyone and cause a race war now at the same time.

:+1:

RemieRichards commented on code/datums/diseases/doxpox.dm:
```diff
@@ -0,0 +1,30 @@
+	switch(stage)
+		if(1)
+			affected_mob.equip_to_slot_or_del(new /obj/item/clothing/suit/armor/reactive/pox/first(affected_mob), SLOT_OCLOTHING)
+			affected_mob.visible_message("<span class='danger'>[affected_mob] removes their outer clothing, revealing their pox covered hands!</span>")
```
>armour
>references hands

RemieRichards commented on code/datums/diseases/doxpox.dm:
```diff
@@ -0,0 +1,30 @@
+	..()
+	switch(stage)
+		if(1)
+			affected_mob.equip_to_slot_or_del(new /obj/item/clothing/suit/armor/reactive/pox/first(affected_mob), SLOT_OCLOTHING)
```
it's ```slot_wear_suit``` not ```SLOT_OCLOTHING```

RemieRichards commented on code/datums/diseases/doxpox.dm:
```diff
@@ -0,0 +1,30 @@
+			affected_mob.visible_message("<span class='danger'>[affected_mob] removes their outer clothing, revealing their pox covered hands!</span>")
+			affected_mob.say( pick( list("Arrrggh!", "It hurts!", "Don't touch me!", "Stay back! Don't touch me!") ) )
+		if(7)
+			qdel(SLOT_OCLOTHING)
```
it's ```slot_wear_suit``` not ```SLOT_OCLOTHING```

RemieRichards commented on code/datums/diseases/doxpox.dm:
```diff
@@ -0,0 +1,30 @@
+			affected_mob.visible_message("<span class='danger'>[affected_mob] poxes get worse, and looks even more painful!</span>")
+			affected_mob.say( pick( list("ARRRRRRGHHHHH!", "The pain! IT SPEAKS TO ME", "STAY BACK!.", "EVERYTHING HURTS", "MAKE IT STOP") ) )
+		if(14)
+			qdel(SLOT_OCLOTHING)
```
it's ```slot_wear_suit``` not ```SLOT_OCLOTHING```

LanCartwright committed:
Adds Asthmosthia

Victim creates atmospherics gases.

LanCartwright committed:
Changes some existing symptoms

Spontaneous combustion
Increased resistance and stage speed from -4 to -3

Necrotizing Fasciitis
Resistance from -4 to -3
Transmittable from -4 to -2

Sneezing
Stealth from -2 to -1
Resistance from 3 to 2

Hyphema
Resistance from -4 to -2

Eternal Youth
Resistance and stage speed from 4 to 5

LanCartwright committed:
Adds Scalerot

Targets lizards only.
Causes deadly clone damage due to loss of scales, before rotting their scales off all together and reverting them to a humanoid form.

LanCartwright committed:
Sensory restoration

Victim passively creates inacusiate, antihol, syanptizine, and mannitol.

LanCartwright committed:
Adds Viral adaptation

Massively buffs stealth and resistance, reduces stage speed.

LanCartwright committed:
Adds Bubonic Scabies

A human only virus

LanCartwright committed:
Adds dox pox

Heals self but prevents wearing clothing
Touching the infected injures them

LanCartwright committed:
Adds Kilkof Hemorrhagic fever

21 stage virus as a test for later use.
Slowly incubates, eventually starts killing you. Once stage 21 hits, you basically die.

LanCartwright committed:
Adds Andromeda blobitis

Slowly converts infected to blob zombies

LanCartwright committed:
Skin mutation

Turns victim into lizard. Non-lizards only.

LanCartwright committed:
Nerfs Fungal Tuberculosis

Removes blood vomiting - you can now be immune by staying away from purposely,
or accidentally made blood pools and using internals/sterile mask.

LanCartwright committed:
HOLD THE PHONE

KazeEspada commented on code/datums/diseases/advance/symptoms/lesserzombie.dm:
```diff
@@ -0,0 +1,54 @@
+	spread_text = "Consumption of or contact with invasive organic material"
+	cure_text = "Phenol and Mannitol, or Atropine"
+	cures = list("phenol", "mannitol")
+	cures = list("atropine")
```
cures defined twice.

LanCartwright replied:
identional. if not done so, you will require both chems to cure.

RemieRichards replied:
it literally doesn't work like that.

Giacom replied:
If you set a field to a variable twice, which was inherited, the last value that BYOND reads will be set to that. In this case, BYOND will read line 6, set the value, and then read line 7 and replace what was set in line 6.

KazeEspada commented on code/datums/diseases/advance/symptoms/bubonicscabies.dm:
```diff
@@ -0,0 +1,40 @@
+	cures = list("cleaner")
+	cures = list("sterilizine")
+	agent = "Cosmic Sarcoptes scabiei bubonicus"
+	viable_mobtypes = list(/mob/living/carbon/human, /mob/living/simple_animal/mouse)
```
As much as the idea is great, I don't think mice can actually carry diseases.

KazeEspada commented on code/datums/diseases/advance/symptoms/bubonicscabies.dm:
```diff
@@ -0,0 +1,40 @@
+	cure_text = "Sterilizine or Space cleaner"
+	cure_chance = 33
+	cures = list("cleaner")
+	cures = list("sterilizine")
```
Defined cures twice.

LanCartwright replied:
ditto

RemieRichards replied:
it literally doesn't work like that.

KazeEspada commented on code/datums/diseases/advance/symptoms/lizarditis.dm:
```diff
@@ -0,0 +1,40 @@
+ 				affected_mob.adjustBruteLoss(20)
+ 				affected_mob.reagents.add_reagent("itching_powder", 15)
+			if(7)
+				affected_mob.dna.species = new /datum/species/lizard()
```
change species proc.

LanCartwright replied:
what's wrong with it?

KazeEspada commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,45 @@
+					M << "<span class='notice'>[pick("Your scales rot away and reveal flesh")]</span>"
+					M.adjustCloneLoss(45)
+				if(5)
+					M.dna.species = new /datum/species/human()
```
change species proc

LanCartwright replied:
ditto

KazeEspada commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,45 @@
+					M.update_icons()
+					M.update_hair()
+					M << "<span class='notice'>[pick("You tear at your scales and rip off your scaley skin!")]</span>"
+					M.adjustCloneLoss(200)
```
Lizardtitis doesn't kill you or bubonic plague. Why does this one?

LanCartwright replied:
scalerot is deadlier than bubonic plague

LanCartwright committed:
bunch of files got messed up

LanCartwright committed:
Merge branch 'newviruses' of https://github.com/LanCartwright/-tg-station into newviruses

LanCartwright commented:
I amm now entirely drunk

here goes

_disease.dm
Adds var/needs_all_cures
Viruses can now have more than one single chemical cure.
To enable this, add needs_all_cures = FALSE to your virus (default TRUE).

Added asthmosthia.dm
Virus will create atmospheric gases.

fire.dm
Buffed ressitance and stage speed.

flesh_eating.dm
Buffed transmittance and resistance.

Added scalerot.dm
Causes clone damage to lizards only.
If a lizard dies, the remainder of their flesh rots away, leaving them just a humanoid corpse.

Added sensory_restoration.dm
Infected creates inacusiate, antihol, synaptizine, and mannitol passively.

sneezing.dm
Buffed stealth and resistance. Subject to revert or change.

Adds viraladaptation.dm
Virus will adapt and become stealthier and better at resisting cures.
Stage speed decreases in return.

vision.dm
Buffs Hyphema resistance.

youth.dm
Buffs resistance and stage speed.

Added bubonicscabies.dm
Human only virus.
Causes the infected to massively scratch and itch themselves, causing sores and damage.

Added doxpox.dm
Causes the infected to be unable to wear outer clothing.
Accosting the infected will burst their poxes, causing injury.
Will cure itself automatically.

Added kilkofhemorrhage.dm
21 stage virus that slowly incubates in the infected.
Stages 16 to 20 causes massive crippling symptoms.
21 violently kills the infected.

Added lesserzombie.dm
A spore will attempt to control the infected.
Infected will slowly lose control of their body, eventually immediately dying and the hostile blob spore assuming control of the body.

Added lizarditis.dm
Turns any non lizard into lizards.

armour.dm
Adds two pox armours for doxpox virus.

other_reagents.dm
Changes typo in mulligan toxin.
Lye now burns you if eaten or contact with.

optimumtact commented:
Changelog

lordpidey commented:
I worry about lye hurting on splash, as it's very easy to make as a non-chemist, just burn a piece of paper and add water.

KazeEspada commented on code/datums/diseases/doxpox.dm:
```diff
@@ -0,0 +1,30 @@
+			affected_mob.visible_message("<span class='danger'>[affected_mob] removes their outer clothing, revealing their pox covered torso!</span>")
+			affected_mob.say( pick( list("Arrrggh!", "It hurts!", "Don't touch me!", "Stay back! Don't touch me!") ) )
+		if(7)
+			qdel(SLOT_OCLOTHING)
```
NOOOOOOOOOO, use unEquip().

KazeEspada commented on code/datums/diseases/doxpox.dm:
```diff
@@ -0,0 +1,30 @@
+			affected_mob.visible_message("<span class='danger'>[affected_mob] poxes get worse, and looks even more painful!</span>")
+			affected_mob.say( pick( list("ARRRRRRGHHHHH!", "The pain! IT SPEAKS TO ME", "STAY BACK!.", "EVERYTHING HURTS", "MAKE IT STOP") ) )
+		if(14)
+			qdel(SLOT_OCLOTHING)
```
unEquip()

LanCartwright committed:
adds virus containers. updates other viruses

LanCartwright committed:
adds sensory destruction

Buggy123 commented:
You gotta nerf skin mutation to fuck and back. It can kill in less than a minute.

Zilenan91 commented:
Yeah make both of them way less lethal. They kill people dead before they can actually do the effects that they're intended to.

Zilenan91 commented:
I think there's a bug with the lesser zombie virus. I got infected with it and managed to infect 2 other people but we all self-cured. That was either insane RnG or the virus is a bit broken

LanCartwright committed:
adds scarabites and scarabas
changes to lesser zombie and lizarditis
adds virus reagents.
adds scarabite cockroaches

neersighted committed:
Merge remote-tracking branch 'tgstation/master' into newviruses

LanCartwright committed:
minor mistake on scarabite dna check

duncathan commented on code/datums/diseases/advance/symptoms/scalerot.dm:
```diff
@@ -0,0 +1,51 @@
+				if(6)
+					M.adjustCloneLoss(25)
+					M << "<span class='danger'>[pick("You tear at your scales and rip off your scaley skin!")]</span>"
+					if(M.stat == DEAD)
```
why does it only happen when they're dead?

duncathan commented on code/datums/diseases/advance/symptoms/sensory.dm:
```diff
@@ -0,0 +1,103 @@
+			if(4, 5)
+				if (M.reagents.get_reagent_amount("mannitol") < 15)
+					M.reagents.add_reagent("mannitol", 15)
+					M << "<span class='notice'><b>Your mind feels relaxed.</b></span>"
```
as it is, this will make only one chem get added at any given time; higher stages override lower ones. is this what you want?

duncathan commented on code/datums/diseases/advance/symptoms/sensory.dm:
```diff
@@ -0,0 +1,103 @@
+				M.adjustEarDamage(0,30)
+				M << "<span class='warning'><b>Your ear canal hurts!</b></span>"
+			if(2)
+				M.adjustEarDamage(0,30)
```
uh i don't think you meant to have these

duncathan commented:
HOLY SHIT THERE ARE SO MANY VIRUSES HERE

jesus christ maybe like two or three at a time, alright? I'm not reviewing all this shit; split it up

LanCartwright commented:
@duncathan Would you like me to break them down into Virology bitesized updates so they can be merged on an almost individual level?

My apologies, I'll keep large content updates to multiple small updates next time.

duncathan commented:
Yea keep splitting it up thanks
